components/studies/ui/src/main/resources/PhenoTips/StudyClass.xml

```diff
@@ -125,6 +125,46 @@
   <object>
     <name>PhenoTips.StudyClass</name>
     <number>0</number>
+    <className>PhenoTips.StudyVisibilityClass</className>
```

Comment:
Right, I didn't read the rest of the code. But in this case, the configuration shouldn't be here anyway. Don't mix classes with real configuration, it makes upgrading harder.